examples/draw_turtle.rs

```diff
@@ -0,0 +1,314 @@
+
+    turtle.right(86.5);
+    turtle.forward(SIZE * 55.0);
+    turtle.right(86.0);
+
+    // Front leg
+    draw_leg(&mut turtle);
+
+    turtle.right(54.5);
+    turtle.forward(SIZE * 8.0);
+
+    draw_neck(&mut turtle);
+
+    turtle.right(172.6);
+    turtle.forward(SIZE * 40.0);
+    turtle.right(110.0);
+
+    draw_head(&mut turtle);
+
+    turtle.left(128.0);
+    turtle.forward(SIZE * 15.0);
+
+    draw_eye(&mut turtle);
+
+    turtle.set_fill_color(SECOND_COLOR);
+
+    turtle.left(175.0);
+    turtle.forward(SIZE * 43.0);
+
+    draw_right_highlight(&mut turtle);
+
+    turtle.right(18.0);
+    turtle.forward(SIZE * 37.0);
+    turtle.set_heading(180.0);
+
+    draw_middle_highlight(&mut turtle);
+
+    turtle.left(24.0);
+    turtle.forward(SIZE * 36.0);
+    turtle.set_heading(180.0);
+
+    draw_left_highlight(&mut turtle);
+}
+
+fn draw_panther(turtle: &mut Turtle) {
+    turtle.set_fill_color(FIRST_COLOR);
+    turtle.begin_fill();
+
+    for _ in 0..180 {
+        turtle.forward(SIZE);
+        turtle.right(1.0);
+    }
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 3.0);
+        turtle.right(1.0);
+    }
+
+    turtle.set_speed(5);
+    let d = SIZE * 360.0 / std::f64::consts::PI;
+    turtle.forward(d - d / 3.0);
+    turtle.set_speed(10);
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 3.0);
+        turtle.right(1.0);
+    }
+
+    turtle.end_fill();
+}
+
+fn draw_tail(turtle: &mut Turtle) {
+    turtle.set_fill_color(SECOND_COLOR);
+    turtle.begin_fill();
+
+    turtle.left(90.0);
+    turtle.forward(SIZE);
+
+    for _ in 0..45 {
+        turtle.forward(SIZE / 3.0);
+        turtle.left(1.0);
+    }
+
+    turtle.forward(SIZE * 3.0);
+
+    for _ in 0..25 {
+        turtle.forward(SIZE / 3.0);
+        turtle.left(6.0);
+    }
+
+    turtle.forward(SIZE * 9.0);
+    turtle.right(6.0);
+    turtle.forward(SIZE * 8.0);
+
+    turtle.end_fill();
+}
+
+fn draw_leg(turtle: &mut Turtle) {
+    turtle.begin_fill();
+
+    for _ in 0..15 {
+        turtle.forward(SIZE);
+        turtle.left(0.5);
+    }
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 6.0);
+        turtle.left(1.0);
+    }
+
+    turtle.forward(SIZE * 3.0);
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 6.0);
+        turtle.left(1.0);
+    }
+
+    turtle.forward(SIZE * 14.5);
+
+    turtle.end_fill();
+}
+
+fn draw_neck(turtle: &mut Turtle) {
+    turtle.begin_fill();
+
+    for _ in 0..15 {
+        turtle.forward(SIZE * 3.0);
+        turtle.left(1.5);
+    }
+
+    turtle.end_fill();
+    turtle.begin_fill();
+
+    turtle.left(100.0);
+    turtle.forward(SIZE * 20.0);
+    turtle.left(80.0);
+
+    for _ in 0..4 {
+        turtle.forward(SIZE * 3.7);
+        turtle.left(1.5);
+    }
+
+    turtle.left(30.0);
+
+    for _ in 0..27 {
+        turtle.forward(SIZE);
+        turtle.right(1.0);
+    }
+
+    turtle.end_fill();
+}
+
+fn draw_head(turtle: &mut Turtle) {
+    turtle.begin_fill();
+
+    for _ in 0..20 {
```

Comment:
There seem to be a lot similar loops throughout this code. They are all subtly different, so I understand why you can't move them into one function, but could you write some documentation at the top of this file outlining the general technique of what you are doing? I imagine that someone looking at this code (including me) would wonder what each loop is for and how this all works.

The question you want to answer somewhere at the top of this file is: **How are you accomplishing the interesting curves used to draw the turtle in this example?** In the answer to that, you can describe the different loops and how they work together to create the really nice curves in each of the different body parts.